internal/website/content/howto/secrets/_index.md

```diff
@@ -21,3 +22,204 @@ The Go CDK provides access to key management providers in a portable way
+
+1. Decrypt the key from storage using the secret keeper
+2. Use the decrypted key to encrypt or decrypt the message inside your
+   application.
+
+[`secretbox`]: https://godoc.org/golang.org/x/crypto/nacl/secretbox
+
+### Keep Secrets in Configuration {#runtimevar}
+
+Once you have [opened a secrets keeper][] for the secrets provider you want,
+you can use a secrets keeper to access sensitive configuration stored in an
+encrypted `runtimevar`.
+
+First, you create a [`*runtimevar.Decoder`][] configured to use your secrets
+keeper using [`runtimevar.DecryptDecode`][]. In this example, we assume the
+data is a plain string, but the configuration could be a more structured
+type.
+
+{{< goexample src="gocloud.dev/runtimevar.ExampleDecryptDecode" imports="0" >}}
+
+Then you can pass the decoder to the runtime configuration provider of your
+choice. See the [Runtime Configuration How-To Guide][] for more on how to set up
+runtime configuration.
+
+[opened a secrets keeper]: {{< ref "#opening" >}}
+[Runtime Configuration How-To Guide]: {{< ref "/howto/runtimevar/_index.md" >}}
+[`*runtimevar.Decoder`]: https://godoc.org/gocloud.dev/runtimevar#Decoder
+[`runtimevar.DecryptDecode`]: https://godoc.org/gocloud.dev/runtimevar#DecryptDecode
+
+## Supported Services {#services}
+
+### AWS Key Management Service {#aws}
+
+The Go CDK can use customer master keys from Amazon Web Service's [Key
+Management Service][AWS KMS] (AWS KMS) to keep information secret. AWS KMS
+URLs can use the key's ID, alias, or Amazon Resource Name (ARN) to identify
+the key. You can specify the `region` query parameter to ensure your
+application connects to the correct region, but otherwise
+`secrets.OpenKeeper` will use the region found in the environment variable
+`AWS_REGION` or your AWS CLI configuration.
+
+{{< goexample "gocloud.dev/secrets/awskms.Example_openFromURL" >}}
+
+[AWS KMS]: https://aws.amazon.com/kms/
+
+#### AWS Key Management Service Constructor {#aws-ctor}
+
+The [`awskms.OpenKeeper`][] constructor opens a customer master key. You must
+first create an [AWS session][] with the same region as your key and then
+connect to KMS:
+
+{{< goexample "gocloud.dev/secrets/awskms.ExampleOpenKeeper" >}}
+
+[`awskms.OpenKeeper`]: https://godoc.org/gocloud.dev/secrets/awskms#OpenKeeper
+[AWS session]: https://docs.aws.amazon.com/sdk-for-go/api/aws/session/
+
+### Google Cloud Key Management Service {#gcp}
+
+The Go CDK can use keys from Google Cloud Platform's [Key Management
+Service][GCP KMS] (GCP KMS) to keep information secret. `secrets.OpenKeeper`
+will use [Application Default Credentials][GCP credentials]. GCP KMS URLs are
+similar to [key resource IDs][]:
+
+{{< goexample "gocloud.dev/secrets/gcpkms.Example_openFromURL" >}}
+
+[GCP KMS]: https://cloud.google.com/kms/
+[key resource IDs]: https://cloud.google.com/kms/docs/object-hierarchy#key
+
+#### Google Cloud Key Management Service Constructor {#gcp-ctor}
```

Comment:
Robert shortened these ones to GCP Constructor, AWS Constructor, etc in his CL since the full long string already shows up in the parent header.